smithy-build/src/main/java/software/amazon/smithy/build/SmithyBuildImpl.java

```diff
@@ -153,14 +153,37 @@ private static void validatePluginName(String projection, String plugin) {
     SmithyBuildResult applyAllProjections() {
         Model resolvedModel = createBaseModel();
         SmithyBuildResult.Builder builder = SmithyBuildResult.builder();
+        Map<String, ProjectionConfig> serialProjections = new TreeMap<>();
+        Map<String, ProjectionConfig> parallelProjections = new TreeMap<>();
         config.getProjections().entrySet().stream()
```

Comment:
Can you leave a comment that describes the what and why of this branching? I'd also like to see a TODO item noted here that we could refactor this to be much more efficient by running all parallelizable plugins across all projections in parallel followed by all serial plugins. As each plugin completes, the SmithyBuildResult.Builder would be updated. This is a bigger refactor and all hidden as an implementation detail, which is why I'm fine delaying it.